src/table.cpp

```diff
@@ -76,6 +76,7 @@ std::vector<JoinResult> Table::hashJoin(size_t index, Table &other, size_t other
     using map_t = std::unordered_map<uint32_t, size_t, std::hash<uint32_t>, std::equal_to<uint32_t>, allocator_t>;
 
     // Allocate map on second column (usually remote)
+    // TODO: I think the first parameter shouldn't be 0 (hoping to let the map use its internal default)
     map_t col_map (0, std::hash<uint32_t>(), std::equal_to<uint32_t>(), allocator_t(other.column(other_index)->numaNode()));
```

Comment:
After removing this from the `unordered_multimap` below, it worked ***a lot*** faster, I think.